Improve the getic.sh to retrive restart files for warm start for GEFS

XianwuXue-NOAA opened the pull request:
**Description**

This PR includes two sets of changes related to merging GEFS into the GFS system:
1) Change the DATA to include jobid to avoid different members using same DATA
2) Add new features to retrieve GFS and ensemble members restart files and stage them in the appropriate directories

<!-- Please provide reference to the issue this pull request is addressing. -->
<!-- For e.g. Fixes #IssueNumber -->

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

The script was tested on Hera.

<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
<!--
- [ ] Clone and Build tests on WCOSS Dell P3
- [ ] Cycled test on Orion
- [ ] Forecast-only test on Hera
-->
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

Refs: #911

XianwuXue-NOAA committed:
Improve the getic.sh to retrive restart files for warm start

 On branch feature/gefs_v13_911_getic_warm_start
 Changes to be committed:
	modified:   jobs/rocoto/getic.sh

Refs: #911

aerorahul commented on jobs/rocoto/getic.sh:
```diff
@@ -82,11 +82,134 @@ if [[ $gfs_ver = "v16" && $EXP_WARM_START = ".true." && $CASE = $OPS_RES ]]; the
+                mv $f $fnew
+            done
+            cd RESTART
+            module load nco/4.9.3
```
The `nco` module is loaded in `module_base.$target`.

XianwuXue-NOAA replied:
OK, I will delete this line.

aerorahul commented on jobs/rocoto/getic.sh:
```diff
@@ -82,11 +82,134 @@ if [[ $gfs_ver = "v16" && $EXP_WARM_START = ".true." && $CASE = $OPS_RES ]]; the
+                mv $f $fnew
+            done
+            cd RESTART
+            module load nco/4.9.3
```
same comment as above.

XianwuXue-NOAA replied:
No problem. I will delete this one.

aerorahul commented on jobs/rocoto/getic.sh:
```diff
@@ -82,11 +82,134 @@ if [[ $gfs_ver = "v16" && $EXP_WARM_START = ".true." && $CASE = $OPS_RES ]]; the
+            elif [ $cmem -gt 10 ]; then
+                sgrp=2
+            elif [ $cmem -gt 0 ]; then
+                sgrp=1
```
seems like this can be reduced to `cmem/10` and/or `cmem%10` rather than this big `if-else` block.

aerorahul reviewed (changes requested):
In principle the changes are fine.
A few minor comments apart from those in the review:
- remove the `echo` where unnecessary and all the `ls` statements.
- module loads need to go out from this script

A major comment:
- a lot of the code is duplicated.  Can we work to reduce that?

XianwuXue-NOAA commented:
@aerorahul I am working on reducing the duplicated statements?

XianwuXue-NOAA committed:
Optimized the script

 On branch feature/gefs_v13_911_getic_warm_start
 Changes to be committed:
	modified:   jobs/rocoto/getic.sh

Refs: #911

XianwuXue-NOAA commented:
@aerorahul I have modified the script and pushed to this PR

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_911_getic_warm_start
